### ISSUE TITLE
Discussion: what quote styles should be included?

### ISSUE THREAD

**Issue by missmatsuko** [missmatsuko/localize.css#8]
Presumably, these styles for `<q>` should already be in user-agent stylesheets:
https://html.spec.whatwg.org/multipage/rendering.html#quotes

I think there is no need to duplicate user-agent styles, since we can build on top of what's already there. In the case of older browsers, we could include the styles that are in the latest user-agent stylesheets, and then add our extended styles. 

However, I have not figured out what's is actually included in the latest user-agent stylesheets, as they don't seem to exactly follow what's in that spec.

I have observed additional styles not explicitly stated in the above in my current browser (Edge 81.0.416.72, which is a Chromium browser).

For example, there are no styles for de-CH in that stylesheet. I would have thought that meant it would default to the styles for de (maybe this is incorrect, and it actually inherits from the country...?), but I saw different quote styles for it in Edge.

Here is a screenshot of what I see in Edge, and a JS Fiddle with its code:
<img width="186" alt="Screenshot showing different quote styles for de and de-CH." src="https://user-images.githubusercontent.com/17884198/82248883-bd66cc00-98fd-11ea-966c-fa6f24027b40.png">
[JS Fiddle showing quote styles for de and de-CH](https://jsfiddle.net/ba0x92nr/).

**Comment by missmatsuko** [comment]
For reference, locations of user-agent stylesheets (via [this StackOverflow answer](https://stackoverflow.com/a/55215684)). 

These don't actually include the quote styles directly (just open-quote and close-quote), so they must be defined elsewhere.

- [Gecko (Firefox)](https://dxr.mozilla.org/mozilla-central/source/layout/style/res/html.css)
- [Chromium (Chrome, Edge?)](https://chromium.googlesource.com/chromium/src/third_party/+/master/blink/renderer/core/html/resources/html.css)
- [WebKit (Safari)](https://trac.webkit.org/browser/trunk/Source/WebCore/css/html.css)

**Gecko**
Generated quotes file in Gecko:
https://dxr.mozilla.org/mozilla-central/source/intl/locale/cldr-quotes.inc

This is the file that generated that previous file. Looks like they are getting data from the CLDR, which does not have different quote styles for de-CH:
https://dxr.mozilla.org/mozilla-central/source/intl/locale/cldr-quotes.pl

And resulting render in FF corroborates this:
![Screenshot of same quote styles for de and de-CH.](https://user-images.githubusercontent.com/17884198/82254934-9e216c00-9908-11ea-9238-7d15957fa988.jpeg)

**Chromium**
Found this file for quote styles in Chromium, which **does** include de-CH:
https://chromium.googlesource.com/chromium/src/third_party/+/master/blink/renderer/core/layout/layout_quote.cc

Funnily enough, it references its source as that HTML Standard link, which **does not** include de-CH??

**WebKit**
Looks to do the same as Chrome:
https://trac.webkit.org/browser/trunk/Source/WebCore/rendering/RenderQuote.cpp

**Comment by missmatsuko** [comment]
I *think* this is the source of the quote marks for the styles in [the HTML standard link](https://html.spec.whatwg.org/multipage/rendering.html#quotes):
https://github.com/unicode-org/cldr/tree/master/common/main

Looks like quote styles for de-CH were once in the HTML standard, [added here](https://github.com/whatwg/html/commit/266660173c1f83aafa1cf74077be58cce271e054) and [removed here](https://github.com/whatwg/html/commit/07084548449afb4a3a867cc2782f91095bae1e89).

CLDR used to have different quotes for de-CH, [but that was removed here](https://github.com/unicode-org/cldr/commit/b2720055ed0efeef324531de03b349cff91947dd). Why? And which is correct? Are both acceptable?

**Comment by tschach** [comment]
Hi, the guillemets «» quote style, like in France, is definitely the standard style in Switzerland. It is mandatory for official documents of the federal government, all major newspapers are using it. I cannot say anything on the CLDR. Maybe that helps?

**Comment by missmatsuko** [comment]
Thanks, @tschach!

Yeah I am finding the CLDR stuff very confusing. I'm trying to figure out if it was removed by mistake or insufficient data or the data actually leaned towards German-style quotes...? Will try to look into it more.

**Comment by tschach** [comment]
You're welcome. Let me know if you need more input on this.

**Comment by jaller94** [comment]
I didn't know that user-agent styles already cover some of these. Good to know.

For a long time my computers are configured to use the (Canadian) English language with German (Germany) locale. I have no idea what impacts that has on the styles in my browsers. My contributions to the German (Germany) locale was based on school teachings and the Wikipedia page for the correct Unicode characters.

As I've never lived in Switzerland, I don't feel qualified to answer the question about their quotation marks.

**Comment by missmatsuko** [comment]
For now, let's leave quote styles for de out since it's in the CLDR and supposed to be in user-agents, and keep de-CH in since it has a reference and an advocate, which is more vetting than what I've added already haha.

**Comment by tschach** [comment]
Then the change https://github.com/missmatsuko/localize.css/pull/7/commits/7392b7ab8d0ad234012bcc3ce3bb0565902b3d8f#diff-4df412e64cc47d215fac91b247eb3917 for `de-CH` in https://github.com/missmatsuko/localize.css/pull/7 would also apply to `fr-CH` and `it-CH`. And if I am not wrong, the existing `rm` is the fourth Swiss language, Rhaeto-Romanic? If so, it already has the correct quote style for Switzerland.

**Comment by missmatsuko** [comment]
Looks like fr-CH is in the [user-agent stylesheet](https://html.spec.whatwg.org/multipage/rendering.html#quotes) but it-CH could be added. rm I think is for Romansh?

**Comment by tschach** [comment]
Yes, Romansh is the proper name, thanks. So I could make a PR for `it-CH`?

**Comment by missmatsuko** [comment]
Sure, thank you!
